### PR TITLE
Ux0.2

### PR DESCRIPTION
**Pull request by faultyserver**
Version 0.2 of the UI. Nothing really changed, but it's more feature complete.

### CODE REVIEW

File: dli_app/templates/reports/all.html [hunk 1]
```diff
@@ -20,21 +24,38 @@
...
+                </button>
+              </form>
+
+              <form method="POST" action="{{ url_for('reports.delete_report', report_id=report.id) }}">
+                <button type="submit" class="btn btn-danger btn-xs">Delete</button>
+              </form>
+
+              <div class="btn-group">
+                <a href="{{ url_for('reports.view_report', report_id=report.id) }}" class="btn btn-info btn-xs">View</a>
+                <a href="{{ url_for('reports.submit_report_data', dept_id=current_user.dept_id) }}" class="btn btn-default btn-xs">Submit Data</a>
```
Comment (gorel) [inline dli_app/templates/reports/all.html]: dept_id parameter is not needed.
At least in this sprint, we don't really have a way of submitting data on a per-report basis. It only works on a site-wide scale. You could actually put the "Submit report data" button at the top of the page if you want.

File: dli_app/templates/reports/all.html [hunk 2]
```diff
@@ -20,21 +24,38 @@
...
```
Comment (gorel) [inline dli_app/templates/reports/all.html]: Same thing as above.

File: dli_app/templates/reports/me.html [hunk 3]
```diff
@@ -28,46 +32,60 @@
...
+                </button>
+              </form>
+
+              <form method="POST" action="{{ url_for('reports.delete_report', report_id=report.id) }}">
+                <button type="submit" class="btn btn-danger btn-xs">Delete</button>
+              </form>
+
+              <div class="btn-group">
+                <a href="{{ url_for('reports.view_report', report_id=report.id) }}" class="btn btn-info btn-xs">View</a>
+                <a href="{{ url_for('reports.submit_report_data', dept_id=current_user.dept_id) }}" class="btn btn-default btn-xs">Submit Data</a>
```
Comment (gorel) [inline dli_app/templates/reports/me.html]: Same as above.

File: dli_app/templates/reports/me.html [hunk 4]
```diff
@@ -28,46 +32,60 @@
...
-    </table>
-  </div>
+
+              <form method="POST" action="{{ url_for('reports.delete_report', report_id=report.id) }}">
+                <button type="submit" class="btn btn-danger btn-xs">Delete</button>
+              </form>
+
+              <div class="btn-group">
+                <a href="{{ url_for('reports.view_report', report_id=report.id) }}" class="btn btn-info btn-xs">View</a>
+                <a href="{{ url_for('reports.submit_report_data', dept_id=current_user.dept_id) }}" class="btn btn-default btn-xs">Submit Data</a>
```
Comment (gorel) [inline dli_app/templates/reports/me.html]: Last time.

**Comment by faultyserver** [comment]
@gorel btw, the admin pages all give TypeErrors.

**Comment by gorel** [comment]
image

A couple minor suggestions, but not huge:
Can we add these buttons to the "favorite reports" list:
- unfavorite
- delete (if report.user.id == current_user.id will check that this is allowed)

And idk if it's possible to space out those buttons nicer, but they all sort of run together and look odd right now.

Finally, I had the suggestion about moving submit_data to the top and not passing in dept_id

**Comment by gorel** [comment]
Admin pages haven't been set up yet. Admin pages will work correctly after https://github.com/gorel/dli-reports/pull/43 has landed

**Comment by faultyserver** [comment]
In short, yes, but I'm a bit lazy right now. I won't have time to add them before tomorrow, but I'll definitely get everything in next week.

**Comment by gorel** [comment]
Sounds good. Also, can you set it so the delete button only shows up if the user owns the report?

The back end will give the user an error otherwise. It fails gracefully and gives the user a message, but it's misleading to show them the delete button when they don't have permission.

Other than this,

LGTM.

**Comment by ad08fee3** [comment]
[redacted]

**Pull request merged**